internal/api/impl/auth/service.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"errors"
+
+	"github.com/perses/perses/internal/api/interface/v1/user"
+	databaseModel "github.com/perses/perses/internal/api/shared/database/model"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+)
+
+// arbitrate decides if we take the old or new string.
+// Return a boolean saying if the result is different from old value.
+func arbitrate(old, new string) (string, bool) {
```

Comment:
```suggestion
func isNew(old, new string) (string, bool) {
```

Reply:
If it was only a bool returned I would agree. But I agree also that arbitrate is not good 

What if I rename it `chooseNewIfPresent` ?

Reply:
yup that works for me !

Reply:
I renamed it to `useNewIfPresent`